specifyweb/specify/views.py

```diff
@@ -391,3 +392,84 @@ def set_admin_status(request, userid):
     else:
         user.clear_admin()
         return http.HttpResponse('false', content_type='text/plain')
+
+@openapi(schema={
+    'post': {
+        "requestBody": {
+            "required": True,
+            "description": "Replace a new agent for an old agent.",
+            "content": {
+                "application/x-www-form-urlencoded": {
+                    "schema": {
+                        "type": "object",
+                        "description": "The error.",
+                        "properties": {
+                            "old_agent_id": {
+                                "type": "integer",
+                                "description": "The old AgentID value of the agent record that is to be replaced by the new one."
+                            },
+                            "new_agent_id": {
+                                "type": "integer",
+                                "description": "The new AgentID value of the agent that is replacing the old one."
+                            }
+                        },
+                        'required': ['old_agent_id', 'new_agent_id'],
+                        'additionalProperties': False
+                    }
+                }
+            }
+        },
+        "responses": {
+            "204": {"description": "Success",},
+            "400": {"description": "The AgentID specified does not exist."}
+        }
+    },
+})
+@login_maybe_required
+@require_POST
+def agent_record_replacement(request: http.HttpRequest, old_agent_id, new_agent_id: int) -> http.HttpResponse:
+    """Replaces all the foreign keys referencing the old AgentID
+    with the new AgentID, and deletes the old agent record.
+    """
+    check_permission_targets(request.specify_collection.id, request.specify_user.id, [])
+
+    # Create database connection cursor
+    cursor = connection.cursor()
+
+    with transaction.atomic():
+        # Check to make sure both the old and new agent IDs exist in the table
+        cursor.execute("SELECT * FROM specify.agent WHERE AgentID = %s;", [old_agent_id])
+        if len(cursor.fetchall()) == 0:
+            return http.HttpResponseBadRequest("AgentID: " + old_agent_id + " does not exist.")
+        cursor.execute("SELECT * FROM specify.agent WHERE AgentID = %s;", [new_agent_id])
+        if len(cursor.fetchall()) == 0:
+            return http.HttpResponseBadRequest("AgentID: " + new_agent_id + " does not exist.")
+
+        # Get all of the columns in all of the tables of specify the are foreign keys referencing AgentID
+        sql_get_cols_ref_agent_id = """
+        SELECT TABLE_NAME, COLUMN_NAME
+        FROM INFORMATION_SCHEMA.KEY_COLUMN_USAGE
+        WHERE
+        REFERENCED_TABLE_SCHEMA = 'specify' AND
```

Comment:
Here there `REFERENCED_TABLE_SCHEMA` should be the name of the database that Specify 7 is using. It won't be "specify" in general. You could get it from the django.conf.settings DATABASE_NAME variable. Or there might be a preferred way to get it in Django. I would google something like "get current database name in Django".